tests/algorithms/test_preference_comparisons.py

```diff
@@ -171,10 +172,15 @@ def test_trajectory_dataset_shuffle(
         cartpole_expert_trajectories,
         rng,
     )
-    sample = dataset.sample(num_steps)
-    sample2 = dataset.sample(num_steps)
+    n_trajectories = len(cartpole_expert_trajectories)
+    flakiness_prob = 1 / n_trajectories
+    max_flakiness = 1e-6
+    # Choose num_samples s.t. flakiness_prob**num_samples <= max_flakiness
+    num_samples = math.ceil(math.log(max_flakiness) / math.log(flakiness_prob))
+    samples = [dataset.sample(num_steps) for _ in range(num_samples)]
```

Comment:
This makes sense to me -- personally it's slightly more intuitive to resample iff the trajectories are equal rather than generating all the samples at the start but num_samples is small enough that I don't think it makes a different.

Reply:
Good point, changed.